common/src/app/common/types/file.cljc

```diff
@@ -116,33 +116,105 @@
   ([libraries library-id component-id]
    (ctkl/get-component (dm/get-in libraries [library-id :data]) component-id)))
 
+(defn get-component-library
```

Comment:
Added a bunch of helpers here to manage components reading their content from the corresponding page of a file (or the component itself if still in v1)